src/app/containers/ContactsContainer.js

```diff
@@ -160,59 +176,37 @@ const ContactsContainer = ({ location, history }) => {
                         {c('Warning').ngettext(
                             msgid`This action will permanently delete the selected contact. Are you sure you want to delete this contact?`,
                             `This action will permanently delete selected contacts. Are you sure you want to delete these contacts?`,
-                            activeIDs.length
+                            filteredCheckedIDs.length
                         )}
                     </Alert>
                 </ConfirmModal>
             );
         });
         const clearChecked = () => {
             setCheckedContacts(Object.create(null));
-            setCheckAll(false);
         };
-        if (checkAll && !contactGroupID) {
+        if (checkedContacts.length === contacts.length) {
```

Comment:
Actually I was writing the wrong condition here. It should be `filteredCheckedIDs.length === contacts.length`, otherwise we could maybe run into the problem of clearing all contacts inadvertently because I want to erase just all the contacts that I see (filtered ones), but the checked ones are all of them. Thanks for pointing it out!!